### ISSUE TITLE
Add ReadArrayList, WriteArrayList to File methodmap

### ISSUE THREAD

**Issue by lazyuselessman** [alliedmodders/sourcemod#1687]
# Suggestion
Add new two methods for `File` methodmap.
# ReadArrayList
Read from file num_items * size bytes and set them as `ArrayList` data.
```
ReadArrayList(ArrayList items, int num_items, int size, int offset = 0)
```
items - ArrayList handler;
num_items - amount items to read from File;
size - ArrayList blocksize;
offset - item to start from.

basically (pseudo code in c)
```
fread((data_pointer + offset * size), size, num_items, file_ptr); 
```
# WriteArrayList
Write from ArrayList from offset num_items amount into file.
```
WriteArrayList(ArrayList items, int num_items, int size, int offset = 0)
```
items - ArrayList handler;
num_items - amount items to write from ArrayList;
size - ArrayList blocksize, (could be omitted);
offset - item to start from.

basically (pseudo code in c)
```
fwrite((data_pointer + offset * size), size, num_items, file_ptr); 
```

As far as `ArrayList` is single `&m_Data[m_Size * m_BlockSize]` it could be written, read like that.

# Why
To not perform [this](https://github.com/shavitush/bhoptimer/blob/0c91c19ba64c6380f402605b62bb57ddb2da863c/addons/sourcemod/scripting/include/shavit/replay-file.inc#L363). By this i mean convert hidden `m_Data` into `int[]` and only after dump it into file when it could be dumped into file directly.
# 
Suggested functions are actually same as
```
	public native int Read(int[] items, int num_items, int size);
	public native bool Write(const int[] items, int num_items, int size);
```
but read / write will be performed from hidden `m_Data` of `ArrayList`, what itself is an array with fixed size.

**Comment by lazyuselessman** [comment]
Or as far as you are using read / write by byte only
```c
	size_t Read(void *pOut, int size) override {
		return fread(pOut, 1, size, fp_);
	}
	size_t Write(const void *pData, int size) override {
		return fwrite(pData, 1, size, fp_);
	}
```

for read it will be
```
fread((data_pointer + offset * size), 1, size*num_items, file_ptr); 
```
and for write
```
fwrite((data_pointer + offset * size), 1, size * num_items, file_ptr); 
```

but this is worse than previous because it is possible to overflow on `size * num_items`.

**Comment by lazyuselessman** [comment]
No longer relevant.
